core/src/main/scala/org/ensime/indexer/FileWatchers.scala

```diff
@@ -155,3 +167,116 @@ private class ApachePollingFileWatcher(
     fm.stop()
   }
 }
+
+object JarJava7Watcher {
+  lazy val jarWatcher = new Java7Watcher()
```

Comment:
yes that's not ideal, but to reduce the number of threads we do need a small number of global Java7 WatchServices. I din't find a better way to pass them to `SourceWatcher` and `ClassfileWatcher` classes.